rpc/harmony/v1/types.go

```diff
@@ -390,6 +417,11 @@ func NewStakingTxReceipt(
 		return nil, err
 	}
 
+	effectiveGasPrice := *big.NewInt(0)
+	if receipt.EffectiveGasPrice != nil {
```

Comment:
should have a else case so we can still set the effectivegasprice to gasprice. Not sure what edge case we are going to face if the rpc return an effective gas price of 0

Reply:
Hey @Frozen I've tested against old and new transactions against testnet RPC and found that rn we still have 2 behaviors:
* old transactions have "effectiveGasPrice": 0
* new transactions have effectiveGasPrice==gasUsed

Can we make the same behavior `effectiveGasPrice==gasUsed` for the old transactions?

Reply:
updated to return effectiveGasPrice==gasUsed if no value for effectiveGasPrice